### PR TITLE
fix: guard length-decode int truncation on 32-bit platforms

### PR DESCRIPTION
**Pull request by oleg-kozlyuk-grafana**
## Summary

- Adds a `byteLen > uint64(math.MaxInt)` guard before every `int(byteLen)` conversion in the inline length-delimited header decoder (`emitConsumeBytesLenAt`) and the unknown-field `skipValue` helper.
- Without this, a uint64 length above MaxInt32 silently truncates on GOARCH=386/arm/wasm — bypassing the historical `int(x) < 0` check and the `postIndex > l` bound. Attacker-controlled payloads could advance iNdEx by a small wrap-around offset and continue parsing into the malicious tail.
- The new guard subsumes the historical `int(x) < 0` check (byteLen ≤ MaxInt ⇒ int(byteLen) ≥ 0 on every supported GOARCH), so it replaces the redundant branch rather than stacking on top.

Closes wiresmith-vk8 (SEC-3, P0).

## Test plan

- [x] `go test ./test/... ./compiler/...` (all pass)
- [x] `GOOS=linux GOARCH=386 go build ./...` (compiles cleanly under 32-bit)
- [x] `GOOS=linux GOARCH=386 go vet ./...` (clean)
- [x] New unit test `TestUnmarshalRejectsLengthAboveMaxInt` covers both the known-field and unknown-field paths, plus the MaxInt boundary case
- [x] `go test ./test/fuzz/ -fuzz=^FuzzUnmarshal$ -fuzztime=10s` (no crashes; ~700k execs)
- [x] Benchmarks: regenerated unmarshal/size benchmarks show ±1–3 % movement, within the same-run noise floor (SizeTraces moved -2 % on an unchanged code path)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: compiler/types/type.go [hunk 1]
```diff
@@ -224,8 +225,13 @@ func emitConsumeBytesLenAt(e Emitter, indent string) {
 	e.Writef("%s\tbyteLen |= uint64(b&0x7F) << shift\n", indent)
 	e.Writef("%s\tif b < 0x80 {\n%s\t\tbreak\n%s\t}\n", indent, indent, indent)
 	e.Writef("%s}\n", indent)
+	// Guard against int truncation on 32-bit platforms (GOARCH=386/arm/wasm).
+	// Without this, a uint64 length above MaxInt32 would silently wrap to a
+	// small positive int and bypass the postIndex>l bound check. The guard
+	// subsumes the historical `int(byteLen) < 0` check: byteLen <= MaxInt
+	// implies int(byteLen) >= 0 on every supported GOARCH.
+	e.Writef("%sif byteLen > uint64(math.MaxInt) {\n%s\treturn io.ErrUnexpectedEOF\n%s}\n", indent, indent, indent)
```
[the remark this answers is not in the document]
Comment (oleg-kozlyuk-grafana) [reply compiler/types/type.go:233]: Checked the upstream library: `google.golang.org/protobuf/encoding/protowire.ConsumeBytes` does exactly this check, just in a fused form:

```go
if m > uint64(len(b[n:])) {
    return nil, errCodeTruncated  // → io.ErrUnexpectedEOF via ParseError
}
```

That single condition subsumes both "length > remaining buffer" and "length > MaxInt" (because `len()` is a non-negative `int`). Upstream maps it to `io.ErrUnexpectedEOF`, so the new guard here is consistent with that.

`proto: negative length` is the gogoproto/vtprotobuf-flavored error for the `intByteLen < 0` path specifically. Since the new guard fires strictly *before* the int conversion — i.e. on a value that's never been a (potentially-negative) int — preserving that error name would be more historical than semantic. I'd rather match the upstream wire-format error than the gogoproto-era one.

Happy to switch if you'd prefer the gogoproto-style `proto: negative length` for parity with neighboring generated code, but I'd lean toward leaving this aligned with upstream.

[the remark this answers is not in the document]
Comment (oleg-kozlyuk-grafana) [reply compiler/types/type.go:233]: New implementation matches protobuf
